pkg/credentialprovider/azure_credentials.go

```diff
@@ -150,16 +152,22 @@ func (a *acrProvider) GetCredentials(ctx context.Context, image string, _ []stri
 	}
 
 	if a.config.UseManagedIdentityExtension {
-		username, password, err := a.getFromACR(ctx, loginServer)
+		username, password, err := a.getFromACR(ctx, targetloginServer)
 		if err != nil {
-			klog.Errorf("error getting credentials from ACR for %s: %s", loginServer, err)
+			klog.Errorf("error getting credentials from ACR for %s: %s", targetloginServer, err)
 			return nil, err
 		}
 
-		response.Auth[loginServer] = v1.AuthConfig{
+		response.Auth[targetloginServer] = v1.AuthConfig{
 			Username: username,
 			Password: password,
 		}
+		if sourceloginServer != "" {
+			response.Auth[sourceloginServer] = v1.AuthConfig{
```

Comment:
you will add auth for both `sourceloginServer` and `targetloginServer`, is that necessary?

Reply:
Yes. I think it does no harm, right.

Reply:
then you will use password pull the mcr image, not sure whether that has side effect, in this case, only `targetloginServer` needs password?

Reply:
When mirror mapping is used, in this case, `sourceLoginServer` is like MCR and `targetloginServer` is like ACR. So, it is the `sourceloginServer` needs password.

Reply:
mcr + acr auth is a MUST, additional auth should bring no hard.
kubelet will send mcr image + acr auth to containerd to pull mcr image, and containerd will mirror the mcr image to acr through [registry host](https://github.com/containerd/containerd/blob/main/docs/hosts.md#registry-host-namespace), and finally pull mcr image from acr server.